### PR TITLE
WIP: Backup API

### PR DESCRIPTION
**Pull request by dschuermann**
WIP: needs testing...

### CODE REVIEW

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/BackupCodeFragment.java [hunk 1]
```diff
@@ -436,10 +457,22 @@ private void startBackup() {
...
                 : Constants.FILE_EXTENSION_ENCRYPTED_BACKUP_PUBLIC);
 
+        Passphrase passphrase = new Passphrase(mBackupCode);
+        if (Constants.DEBUG && mDebugModeAcceptAnyCode) {
+            passphrase = new Passphrase("AAAA-AAAA-AAAA-AAAA-AAAA-AAAA");
+        }
+
+        // special case for remote API, see RemoteBackupActivity
+        if (mReturnBackup) {
+            ((RemoteBackupActivity) getActivity()).handleResult(new CryptoInputParcel(passphrase));
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/BackupCodeFragment.java]: @Valodim Do you have an idea how to handle this separation better?

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/BackupKeyringParcel.java [hunk 2]
```diff
@@ -28,15 +28,12 @@
 
 public class BackupKeyringParcel implements Parcelable {
     public Uri mCanonicalizedPublicKeyringUri;
-    public Passphrase mSymmetricPassphrase;
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/BackupKeyringParcel.java]: @Valodim I will now use CryptoInputParcel for the passphrase. Is this okay or are there reasons to leave this inside the operation parcel?

Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/BackupKeyringParcel.java]: For explanation: the CryptoInputParcels are properly cached when used with the API without further work.

**Comment by dschuermann** [comment]
TODOs for me:
* I wasn't able to finish the backup process when entering the backup code, looks like a problem with the new edittext
* I wasn't able to properly get back to the last character, the special whitecases prevent this
* I need to implement EXTRA_REQUEST_ASCII_ARMOR (currently always true)

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/operations/BackupOperation.java [hunk 3]
```diff
@@ -97,6 +98,12 @@ public BackupOperation(Context context, ProviderHelper providerHelper,
 
     @NonNull
     public ExportResult execute(@NonNull BackupKeyringParcel backupInput, @Nullable CryptoInputParcel cryptoInput) {
+        return execute(backupInput, cryptoInput, null);
```
Comment (Valodim) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/operations/BackupOperation.java]: doesn't this just crash?

**Pull request merged**